dali/imgcodec/decoders/nvjpeg/nvjpeg.cc

```diff
@@ -165,14 +165,16 @@ DecodeResult NvJpegDecoderInstance::DecodeImplTask(int thread_idx,
   DecodingContext ctx = DecodingContext{ resources_[thread_idx] };
   CUDA_CALL(nvjpegDecodeParamsSetOutputFormat(ctx.resources.params, GetFormat(opts.format)));
   CUDA_CALL(nvjpegDecodeParamsSetAllowCMYK(ctx.resources.params, true));
-
+  if (roi.use_roi()) {
+    CUDA_CALL(nvjpegDecodeParamsSetROI(ctx.resources.params, roi.begin[1], roi.begin[0],
+                                       roi.shape()[1], roi.shape()[0]));
+  }
```

Comment:
I think you should reset the params if not using ROI - unless ctx.resources.params is a disposable object.

```suggestion
  } else {
    CUDA_CALL(nvjpegDecodeParamsSetROI(ctx.resources.params, 0, 0, -1, -1));  }
```

Reply:
Wow, nice catch! Fixed